Sources/Layout/MessageCellLayout.swift

```diff
@@ -53,11 +53,11 @@ open class MessageSizeCalculator: CellSizeCalculator {
     public var incomingMessagePadding = UIEdgeInsets(top: 0, left: 4, bottom: 0, right: 30)
     public var outgoingMessagePadding = UIEdgeInsets(top: 0, left: 30, bottom: 0, right: 4)
 
-    public var incomingCellTopLabelAlignment = LabelAlignment.messageLeading(.zero)
-    public var outgoingCellTopLabelAlignment = LabelAlignment.messageTrailing(.zero)
+    public var incomingCellTopLabelAlignment = LabelAlignment(textAlignment: .left, textInsets: .zero)
+    public var outgoingCellTopLabelAlignment = LabelAlignment(textAlignment: .right, textInsets: .zero)
 
-    public var incomingCellBottomLabelAlignment = LabelAlignment.messageTrailing(.zero)
-    public var outgoingCellBottomLabelAlignment = LabelAlignment.messageLeading(.zero)
+    public var incomingCellBottomLabelAlignment = LabelAlignment(textAlignment: .right, textInsets: .zero)
+    public var outgoingCellBottomLabelAlignment = LabelAlignment(textAlignment: .left, textInsets: .zero)
```

Comment:
Migration:
`LabelAlignment.cellCenter(.zero)` = `LabelAlignment(textAlignment: .center, textInsets: .zero)`
`LabelAlignment.cellLeading(.zero)` = `LabelAlignment(textAlignment: .left, textInsets: .zero)`
`LabelAlignment.cellTrailing(.zero)` = `LabelAlignment(textAlignment: .right, textInsets: .zero)`

There's no good replacement for:
`LabelAlignment.messageLeading(.zero)` && `LabelAlignment.messageTrailing(.zero)`

Since these need the `messageContainerView` frame a reference. You can work out all the cases where the alignment is closest to the `AvatarView` because you can use `avatarView.size.width` + `messagePadding.left` for example.

You could always make these computed properties or method that depends on the `messageContainerSize`. I don't think we'll end up caching anything but message container size (if anything at all after this PR) so it should be fair game.

Reply:
I mean this is ok to be honest, most chat UI's aren't aligning things with the trailing edge of the bubble because you can't predict the bubbles width.

Reply:
We don't need to provide `messageLeading`, user can adjust the `textInsets` to fit, it's easy to implement.

Reply:
Closer to complete refactor! 👍

Reply:
Yeah, considering this reduces layout time by 70% it makes caching even less important. Regardless, I still want to merge #579 because the architecture lets users handle custom cells easier and having a less dynamic layout will allow us to explore things like caching/other optimizations if we need to in the future.

Reply:
> We don't need to provide messageLeading, user can adjust the textInsets to fit, it's easy to implement.

I agree. You can do message leading if the avatar is on the left by using `AvatarView.width` + `messagePadding.left` as the inset. 

You can't do this if the avatar is on the right because you don't know how wide the bubble will be. It could be short like `lol` or long like `The quick brown fox`. I think this is fine because users likely don't want to be aligning something this way because it is so volatile.

Reply:
> Yeah, considering this reduces layout time by 70% it makes caching even less important. Regardless, I still want to merge #579 because the architecture lets users handle custom cells easier and having a less dynamic layout will allow us to explore things like caching/other optimizations if we need to in the future.

Yep, a big improvement, caching/other optimization is important when scroll, we can easily add these after these refactor IMO.

>  I think this is fine because users likely don't want to be aligning something this way because it is so volatile.

I agree totally.